process_report/process_report.py

```diff
@@ -335,40 +354,76 @@ def validate_pi_names(dataframe):
 
 def apply_credits_new_pi(dataframe, old_pi_file):
     new_pi_credit_code = "0002"
-    new_pi_credit_amount = 1000
 
     dataframe[CREDIT_FIELD] = None
     dataframe[CREDIT_CODE_FIELD] = None
     dataframe[BALANCE_FIELD] = Decimal(0)
 
-    old_pi_dict = load_old_pis(old_pi_file)
+    old_pi_df = load_old_pis(old_pi_file)
 
-    current_pi_list = dataframe[PI_FIELD].unique()
+    current_pi_set = set(dataframe[PI_FIELD])
     invoice_month = dataframe[INVOICE_DATE_FIELD].iat[0]
+    invoice_pis = old_pi_df[old_pi_df[PI_FIRST_MONTH] == invoice_month]
+    if invoice_pis["Initial Credits"].empty or pandas.isna(
+        new_pi_credit_amount := invoice_pis["Initial Credits"].iat[0]
+    ):
+        new_pi_credit_amount = 1000
+
+    print(f"New PI Credit set at {new_pi_credit_amount} for {invoice_month}")
+
+    one_month_pi_list = old_pi_df[
+        old_pi_df[PI_FIRST_MONTH] == add_invoice_month(invoice_month, -1)
+    ][PI_PI_FIELD]
+    for pi in one_month_pi_list:
+        if pi not in current_pi_set:
+            old_pi_df.loc[
+                old_pi_df[PI_PI_FIELD] == pi, PI_2ND_REMAINDER
+            ] = old_pi_df.loc[old_pi_df[PI_PI_FIELD] == pi, PI_1ST_REMAINDER].iat[0]
```

Comment:
is this possible to break it down into smaller steps?

Reply:
Turns out all this could be a two-liner. Also, because of your suggestion,  when I tested the testcases I realized there was a test case that made the `Apply New PI Credit` test always pass.